### PR TITLE
feat: testnet mode approve sheet

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-####
Figma link (if any):

## What changed (plus any additional context for devs)

we're now showing a sheet making the users

- switch to testnet mode to accept a dapp request if they're connected to a testnet but with testnet mode disabled
- disable testnet mode to accept a dapp request if they're connected to a mainnet but with testnet mode enabled

## Screen recordings / screenshots

https://www.loom.com/share/16f1f9f047c54316bf5409e56db8291d

### CODE REVIEW

File: src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  const dappHost = dappMetadata?.appHost || '';
+
+  const { testnetMode, setTestnetMode } = useTestnetModeStore();
+  const { activeSession } = useAppSession({
+    host: dappHost,
+  });
+
+  const [hint, setHint] = useState<Hint>(INITIAL_HINT);
+
+  console.log('pendingRequest', pendingRequest);
```
Comment (brunobar79) [inline src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx:56]: 🧹 🧹 🧹

**Review by brunobar79 — APPROVED** [review]
This is a really nice ux  🤝

File: src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                  chainName: ChainNameDisplay[hint.chainId],
+                })}
+              </Text>
+              <Text
+                align="center"
+                color="labelTertiary"
+                size="10pt"
+                weight="bold"
+              >
+                🕹️{' '}
```
Comment (derHowie) [inline src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx:111]: guessing the formatter did this, or do you want the space here?

File: src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                <TextLink scale={false} color="green">
+                  {i18n.t('testnet_mode_watcher.testnet_mode')}
+                </TextLink>{' '}
```
Comment (derHowie) [inline src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx:114]: same here

**Review by derHowie — APPROVED** [review]
👍, small comments

File: src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx (hunk 2 above)
Comment (estebanmino) [reply src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx:111]: i want this

File: src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx (hunk 3 above)
Comment (estebanmino) [reply src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx:114]: but not this, will fix thanks

Comment (estebanmino) [reply src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx:114]: actually, i do want it, is a link inside a text

**Pull request merged**